src/odb/src/zutil/util.cpp

```diff
@@ -76,71 +80,47 @@ void RUDYCalculator::setGridConfig(odb::Rect block, int tileCntX, int tileCntY)
   int curX = gridLx;
   for (auto& gridColumn : grid_) {
     gridColumn.resize(tileCntY_);
     int curY = gridLy;
     for (auto& grid : gridColumn) {
-      grid.setRect(curX, curY, curX + tileWidth, curY + tileHeight);
-      curY += tileHeight;
+      grid.setRect(curX, curY, curX + tile_width, curY + tile_height);
+      curY += tile_height;
     }
-    curX += tileWidth;
+    curX += tile_width;
   }
 }
 
 void RUDYCalculator::calculateRUDY()
 {
   // refer: https://ieeexplore.ieee.org/document/4211973
-  const int blockWidth = gridBlock_.dx();
-  const int blockHeight = gridBlock_.dy();
-  const int tileWidth = blockWidth / tileCntX_;
-  const int tileHeight = blockHeight / tileCntY_;
+  const int block_width = gridBlock_.dx();
+  const int block_height = gridBlock_.dy();
+  const int tile_width = block_width / tileCntX_;
+  const int tile_height = block_height / tileCntY_;
 
   for (auto net : block_->getNets()) {
-    // TODO: handle supply nets for routing congestion prediction.
-    if (net->getSigType().isSupply()) {
-      continue;
-    }
-
-    const auto netBox = net->getTermBBox();
-    const auto netArea = netBox.area();
-    if (netArea == 0) {
-      continue;
-    }
-    const auto hpwl = static_cast<float>(netBox.dx() + netBox.dy());
-    const auto wireArea = hpwl * wireWidth_;
-    const auto netCongestion = wireArea / netArea;
-
-    // Calculate the intersection range
-    const int minXIndex
-        = std::max(0, (netBox.xMin() - gridBlock_.xMin()) / tileWidth);
-    const int maxXIndex = std::min(
-        tileCntX_ - 1, (netBox.xMax() - gridBlock_.xMin()) / tileWidth);
-    const int minYIndex
-        = std::max(0, (netBox.yMin() - gridBlock_.yMin()) / tileHeight);
-    const int maxYIndex = std::min(
-        tileCntY_ - 1, (netBox.yMax() - gridBlock_.yMin()) / tileHeight);
-
-    // Iterate over the tiles in the calculated range
-    for (int x = minXIndex; x <= maxXIndex; ++x) {
-      for (int y = minYIndex; y <= maxYIndex; ++y) {
-        Tile& tile = getEditableTile(x, y);
-        const auto tileBox = tile.getRect();
-        if (netBox.overlaps(tileBox)) {
-          const auto intersectArea = netBox.intersect(tileBox).area();
-          const auto tileArea = tileBox.area();
-          const auto tileNetBoxRatio = static_cast<float>(intersectArea)
-                                       / static_cast<float>(tileArea);
-          const auto rudy = netCongestion * tileNetBoxRatio * 100;
-          tile.addRUDY(rudy);
+    if (!net->getSigType().isSupply()) {
+      const auto netBox = net->getTermBBox();
+      processTilesIntersectionSignalNet(netBox, tile_width, tile_height);
+    } else {
+      for (odb::dbSWire* swire : net->getSWires()) {
+        for (odb::dbSBox* s : swire->getWires()) {
+          if (s->isVia()) {
+            continue;
+          } else {
+            odb::Rect wire_rect = s->getBox();
+            processTilesIntersection(wire_rect, tile_width, tile_height, 2);
```

Comment:
Is there a reason for setting the `nets_per_tile` as 2 in this function call?

Reply:
This was a mistake. The number 2 seemed to work better with the macros, and was accidentally added here.

Reply:
Furthermore, I find this solution of adding "nets per tile" (hypothetically assuming their presence) is somewhat inelegant, but it provided a good congestion result on most designs, considering the macros.